src/shared/parameter_v2/config/examples/example_config.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+#include <memory.h>
+#include <stdlib.h>
+
+#include "shared/parameter_v2/config/examples/parameter.h"
+
+///////////////////////////////////////////////////////////////////////
+//                 Generated C Config Structs                        //
+///////////////////////////////////////////////////////////////////////
+//
+// We achieve immutability by setting IntegerParameter_t, BoolParameter_t etc
+// to const. They can only be set during initialization. We also set the ptrs
+// pointing to included configs as const.
+//
+// This allows us to expose the "internals" of the generated FooConfig struct,
+// BarConfig struct, etc.. to let libraries know what parameters and configs
+// are present inside.
+//
+typedef struct FooConfig_s
+{
+    const IntParameter_t* foo_int;
+    const BoolParameter_t* foo_bool;
+} FooConfig_t;
+
+typedef struct BarConfig_s
+{
+    const IntParameter_t* bar_int;
+    const BoolParameter_t* bar_bool;
+} BarConfig_t;
+
+typedef struct ExampleConfig_s
+{
+    // generated from an "include" statement in the yaml,
+    // include:
+    //      - "foo.yaml"
+    //      - "bar.yaml"
+    // It does NOT indicate hierarchy.
+    const FooConfig_t* FooConfig;
+    const BarConfig_t* BarConfig;
+
+    // the parameters defined in example.yaml
+    const BoolParameter_t* example_bool_param;
+    const UIntParameter_t* example_uint_param;
+    const IntParameter_t* example_int_param;
+    const FloatParameter_t* example_float_param;
+    const StringParameter_t* example_string_param;
+} ExampleConfig_t;
+
+typedef struct ThunderbotsConfig
+{
+    const ExampleConfig_t* ExampleConfig;
+} ThunderbotsConfig_t;
```

Comment:
If we're doing the pseudo-class pattern, then these structs should be declared in the header, but defined in the `.c`

Reply:
Did we want to generate `app_dynamic_parameter_getExampleConfig` as well?

the parameter is a psuedo-class, but I think having configs in a pseudo-class will be overkill

Reply:
Oh I see. No I think this is fine.